extensions/camel-k/model-v1alpha1/pom.xml

```diff
@@ -0,0 +1,160 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (C) 2015 Red Hat, Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+            http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+      <groupId>io.fabric8</groupId>
+      <artifactId>camel-k-extension-pom</artifactId>
+      <version>5.3-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>camel-k-model-v1alpha1</artifactId>
+    <packaging>bundle</packaging>
+    <name>Fabric8 :: Camel-K :: Model v1alpha1</name>
+
+    <properties>
+        <osgi.import>
+            io.fabric8.kubernetes.api.builder,
+            !io.fabric8.camelk.*,
+            *
+        </osgi.import>
+        <osgi.export>
+            io.fabric8.camelk.*
+        </osgi.export>
+        <osgi.include.resources>
+          {maven-resources},
+          /camel-k-v1alpha1.properties=target/classes/model.properties
+        </osgi.include.resources>
+    </properties>
+
+    <dependencies>
+      <dependency>
+        <groupId>io.sundr</groupId>
+        <artifactId>builder-annotations</artifactId>
+      </dependency>
+      <dependency>
+        <groupId>io.sundr</groupId>
+        <artifactId>transform-annotations</artifactId>
+      </dependency>
+      <dependency>
+        <groupId>io.fabric8</groupId>
+        <artifactId>kubernetes-model-core</artifactId>
+      </dependency>
+      <dependency>
+        <groupId>io.fabric8</groupId>
+        <artifactId>kubernetes-model-apiextensions</artifactId>
+      </dependency>
+      <dependency>
+        <groupId>io.fabric8</groupId>
+        <artifactId>knative-model</artifactId>
+      </dependency>
+      <dependency>
+        <groupId>io.fabric8</groupId>
+        <artifactId>camel-k-model-v1</artifactId>
+        <version>${project.version}</version>
+      </dependency>
```

Comment:
This dependency seems odd, why does v1alpha1 depend on v1?

Reply:
in camel-k we have some types that have matured from v1alpha to v1, but some CRs have not yet moved to v1

Reply:
Yes, but I read this as the opposite. According to your statement, shouldn't v1 depend on v1alpha1?

Reply:
I think my explanation was not really clear :)

The Flow type is part of v1 but the KameletBinding which is part of v1alpha1 uses Flow hence, v1alpha1 depends on v1

Reply:
:exploding_head:
All clear then :)